### DISCUSSION TITLE
Refresh table on invalidate query with react query

### DISCUSSION THREAD

**Discussion by Moaaz-Hajat** [KevinVandy/material-react-table#183] [Q&A]
Hi

How would one refresh the table when invalidating a query using a table with react query? When I try to invalidate the query with react-query the table does not re-render/refresh.

Thanks for the help.

**Comment by KevinVandy** [comment]
Is your data being memoized? If so, make sure the dependency array is correct.

**Comment by Moaaz-Hajat** [comment]
I am using the example that was in the docs for react query, I have a form that is rendered on the page that I call the ExampleWithReactQueryProvider component. When I click submit on the form, I want to refresh the data with queryClient.invalidateQueries("table-data") . But the data does not refresh. I think it has something to do with when we define the key in the useQuery that includes the filter state variables etc.

**Reply by KevinVandy** [reply]
React Query also has a refetch method. I recently added that to the react-query example
